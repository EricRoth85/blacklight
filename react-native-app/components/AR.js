'use strict';
import React from 'react';
import { View, TouchableHighlight, TouchableOpacity } from 'react-native';
import * as THREE from 'three';
import ExpoTHREE from 'expo-three';
import Expo from 'expo';
import geolib from 'geolib';
console.disableYellowBox = true;

// ---------- NOTE -----------
// This only works with a hard coded memory on local state right now.
// Functionally this is just an AR component that keeps track of a users location,
// the distance, in meters from users current location and the hardcoded memory.
// The sphere is in no way connected to the distance measurement or anything.
// The AR part just renders the sphere always.

export default class AR extends React.Component {

  constructor(props) {
    super(props);
    this.state = {
      memory: {
        title: 'starbucks across the street',
        text: 'Sed ut perspiciatis unde omnis iste natus error sit voluptatem accusantium doloremque laudantium, totam rem aperiam, eaque ipsa quae ab illo inventore veritatis et quasi architecto beatae vitae dicta sunt explicabo. Nemo enim ipsam voluptatem quia voluptas sit aspernatur aut odit aut fugit, sed quia consequuntur magni dolores eos qui ratione voluptatem sequi nesciunt. Neque porro quisquam est, qui dolorem ipsum quia dolor sit amet, consectetur, adipisci velit, sed quia non numquam eius modi tempora incidunt ut labore et dolore magnam aliquam quaerat voluptatem. Ut enim ad minima veniam, quis nostrum exercitationem ullam corporis suscipit laboriosam, nisi ut aliquid ex ea commodi consequatur? Quis autem vel eum iure reprehenderit qui in ea voluptate velit esse quam nihil molestiae consequatur, vel illum qui dolorem eum fugiat quo voluptas nulla pariatur?',
        lng: -74.00882289999998,
        lat: 40.70459220000001,
        authorId: 1,
      },
      distance: NaN,
      intvl: null
    };
  }

  componentDidMount() {
    let intvl = setInterval(() => {
      Expo.Location.getCurrentPositionAsync({ enableHighAccuracy: true })
        .then((result) => {
          let newDistance = geolib.getDistance(
            { latitude: result.coords.latitude, longitude: result.coords.longitude },
            { latitude: this.state.memory.lat, longitude: this.state.memory.lng },
            1,
            3
          );
          this.setState({ distance: newDistance });
        });
    }, 100);

    this.setState({ intvl });
  }

  componentWillUnmount() {
    if (typeof this.state.intvl === 'function') {
      this.state.intvl();
      this.setState({ intvl: null });
    }
  }

  _onGLContextCreate = async (gl) => {
    const arSession = await this._glView.startARSessionAsync();
    const scene = new THREE.Scene();
    const camera = ExpoTHREE.createARCamera(
      arSession,
      gl.drawingBufferWidth,
      gl.drawingBufferHeight,
      0.01,
      1000
    );
    const renderer = ExpoTHREE.createRenderer({ gl });
    renderer.setSize(gl.drawingBufferWidth, gl.drawingBufferHeight);
    let animate;

    scene.background = ExpoTHREE.createARBackgroundTexture(arSession, renderer);

    // This doesn't work at the moment because it only checks initially. And initially this.state.distance
    // is NaN. So it can't compare that to a number. just goes straight to the else.
    // if distance is less than __ then we render the cube/sphere in the scene.
    // Otherwise we don't render the cube/sphere, but still render the scene/camera
    // if (this.state.distance < 100) {
    const geometry = new THREE.SphereGeometry(0.15, 20, 20);
    const material = new THREE.MeshBasicMaterial({ color: 0xee82ee, wireframe: true });
    const cube = new THREE.Mesh(geometry, material);

    cube.position.z = -1.2;
    scene.add(cube);

    animate = () => {
      requestAnimationFrame(animate);
      cube.rotation.x += 0.01;
      cube.rotation.y += 0.01;
      renderer.render(scene, camera);
      gl.endFrameEXP();
    }
    animate();
  }

  render() {
    return (
      <TouchableOpacity onPress={() => console.log('PRESSED')} style={{ flex: 1 }} >
        <Expo.GLView
          ref={(ref) => this._glView = ref}
          style={{ flex: 1 }}
          onContextCreate={this._onGLContextCreate.bind(this)}
        />
      </TouchableOpacity>
    );
  }
}
