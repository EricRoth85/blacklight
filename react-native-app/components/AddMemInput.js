
import React, { Component } from 'react';
import { connect } from 'react-redux';
import { FormLabel, FormInput, Card, Button } from 'react-native-elements';
import { StyleSheet, View, Alert } from 'react-native';
import { Actions } from 'react-native-router-flux';

import { commitMemory } from '../store';

class AddMemInput extends Component {
  constructor(props) {
    super(props);
    this.state = {
      title: 'test title',
      text: 'test memory',
    };
    this.handleTitle = this.handleTitle.bind(this);
    this.handleText = this.handleText.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }
  handleTitle(title) {
    this.setState({ title });
  }

  handleText(text) {
    this.setState({ text });
  }

  doSomething() {
    console.log('fire')
    Actions.memoryListMain()
  }


  handleSubmit() {
    const { latitude, longitude } = this.props;
    const { title, text } = this.state;
    this.setState({ title: '', text: '' });
    this.props.commitMemory(
      {
        title,
        text,
        lat: latitude,
        lng: longitude,
      });
    Alert.alert('Memory Saved!');
    Actions.memoryListMain();
  }


  render() {
    return (
      <View style={styles.container}>
        <Card title="Enter your story">
          <FormLabel>Title</FormLabel>
          <FormInput
            onChangeText={this.handleTitle}
            value={this.state.title}
          />

          <FormLabel>Text</FormLabel>
          <FormInput
            onChangeText={this.handleText}
            value={this.state.text}
          />

          <Button
            small
            backgroundColor="#00BFFF"
            onPress={this.handleSubmit}
            title="submit"
          />
        </Card>
      </View>
    );
  }
}

const mapDispatch = { commitMemory };

export default connect(null, mapDispatch)(AddMemInput);

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: 'white',
  },


});
