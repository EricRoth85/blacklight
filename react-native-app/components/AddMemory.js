import React, { Component } from 'react';
import { StyleSheet, Text, View } from 'react-native';
import { Button } from 'react-native-elements';



export default class NewMemory extends Component {

  render() {
    const { navigate } = this.props.navigation;
    return (
      <View style={styles.container}>
        <Text style={styles.text}>Enter Memory</Text>
        <Button
          onPress={() => navigate('AddMemMap')}
          title="Go to Brent's profile"
        />
      </View>
    )
  }

}


const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: 'green',
    alignItems: 'center',
    justifyContent: 'center',
  },
  text: {
    fontSize: 20,
    textAlign: 'center'
  }
});
