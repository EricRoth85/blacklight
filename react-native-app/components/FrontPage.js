'use strict';
import React, { Component } from 'react';
import { StyleSheet, Text, View, Button } from 'react-native';
// import TabNavigator from 'react-native-tab-navigator';
import { TabNavigator, DrawerNavigator } from 'react-navigation';

import MemoryList from './MemoryList';
import MemoryMap from './MemoryMap';

export default class FrontPage extends Component {

  // static navigationOptions = {
  //   title: 'Blacklight',
  // };

  render() {

    const { navigate } = this.props.navigation;

    return (
      <View style={styles.container}>
        <Text style={styles.text}>THIS IS THE FRONT PAGE</Text>
        <Text style={styles.subText}>THE MAP WILL BE HERE</Text>
        <Button
        onPress={() => navigate('MemoryMap', {})}
        title='Go to Map'
        color='red'
        />

      </View>

    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: 'green',
    alignItems: 'center',
    justifyContent: 'center',
  },
  // text: {
  //   fontSize: 40,
  //   fontWeight: 'bold',
  //   color: 'white',
  //   textAlign: 'center'
  // },
  subText: {
    fontSize: 20,
    fontWeight: 'bold',
    color: 'blue',
  }
});


        // <Button
        //   onPress={() => navigate('MemoryList', {})}
        //   title='Go to memories'
        //   color='red'
        // />
