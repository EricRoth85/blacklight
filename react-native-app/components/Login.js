'use strict';
import React, { Component } from 'react';
import { connect } from 'react-redux';
import { View, ImageBackground, KeyboardAvoidingView } from 'react-native';
import { FormLabel, FormInput, Button, Text } from 'react-native-elements'
import { Actions } from 'react-native-router-flux';
import { emailChanged, passwordChanged, createUserOnServer, getUser } from '../store';


class Login extends Component {

  onEmailChange(text) {
    this.props.emailChanged(text);
  }
  onPasswordChange(text) {
    this.props.passwordChanged(text);
  }
  handleSubmit() {
    const { email, password } = this.props;
    this.props.createUserOnServer({ email, password });
    Actions.root();
  }
  handleLogin() {
    const { email, password } = this.props;
    this.props.getUser({ email, password });
    Actions.root();
  }

  render() {
    return (
      <ImageBackground
        source={require('../images/wallpaper.jpg')}
        style={styles.container}>
        <View style={styles.logoContainer}>
          <Text style={styles.title}>BLACKLIGHT</Text>
        </View>
        <Button style={styles.buttonStyle}
          small
          backgroundColor='#000000'
          onPress={() => Actions.root()}
          title="CONTINUE AS ANONYMOUS"
        />
        <KeyboardAvoidingView style={styles.formContainer} behavior="padding">
          <FormLabel>
            Email
          </FormLabel>
          <FormInput style={styles.input}
            placeholder="Email"
            onChangeText={this.onEmailChange.bind(this)}
          />
          <FormLabel>
            PW
          </FormLabel>
          <FormInput style={styles.input}
            secureTextEntry
            onChangeText={this.onPasswordChange.bind(this)}
          />
          <Button style={styles.buttonStyle}
            small
            backgroundColor='#000000'
            onPress={this.handleLogin.bind(this)}
            title="LOGIN"
          />
          <Button style={styles.buttonStyle}
            small
            backgroundColor='#000000'
            onPress={this.handleSubmit.bind(this)}
            title='SIGNUP'
          />
        </KeyboardAvoidingView>
      </ImageBackground>
    );
  }
}

const mapStateToProps = (state) => ({
  email: state.auth.email,
  password: state.auth.password,
});

const mapDispatchToProps = { emailChanged, passwordChanged, createUserOnServer, getUser };

export default connect(mapStateToProps, mapDispatchToProps)(Login);

const styles = {
  container: {
    flex: 1,
    width: undefined,
    height: undefined,
    backgroundColor: 'transparent',
    alignItems: 'center',
    justifyContent: 'center',
    padding: 20
  },
  title: {
    fontSize: 34,
    fontWeight: 'bold',
    color: 'white',
    textAlign: 'center',
    marginTop: 10,
    fontFamily: 'LaoSangamMN'
  },
  logoContainer: {
    alignItems: 'center',
    flexGrow: 1,
    justifyContent: 'center'
  },
  input: {
    height: 35,
    backgroundColor: 'rgba(192,192,192,0.3)',
    marginBottom: 15
  },
  buttonStyle: {
    padding: 10
  },
  labelContainerStyle: {
    marginTop: 8
  },
  formContainer: {
    padding: 10
  }
};
