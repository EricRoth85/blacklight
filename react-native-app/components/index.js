'use strict';
export Login from './Login';
export Record from './Record';
export Main from './Main';
export RecordInput from './RecordInput';
export NearbyMem from './NearbyMem';
export MemoryList from './MemoryList';
export SingleMemoryView from './SingleMemoryView';
export AR from './AR';

