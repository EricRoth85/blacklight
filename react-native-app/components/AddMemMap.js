

import React, { Component } from 'react';
import { connect } from 'react-redux';
import { FormLabel, FormInput, Card, Button } from 'react-native-elements';
import { StyleSheet, Text, View, AlertIOS } from 'react-native';
import MapView from 'react-native-maps';

import Map from './Map';
import AddMemInput from './AddMemInput';
import { Actions } from 'react-native-router-flux';

import { titleChanged, textChanged, receivedLocation, commitMemory } from '../store';

export default class AddMemory extends Component {
  constructor(props) {
    super(props);
    this.state = {
      latitude: 0,
      longitude: 0,
      disableButton: true,
    }
  }

  componentDidMount() {
    AlertIOS.alert(
      'Choose Location',
      'use your current location or drop a pin to set location',
      [
        {
          text: 'use my location', onPress: () => {
            Actions.addMemInput(this.state)
          }
        },
        { text: 'drop pin' },
      ],
    );
  }


  setMemoryLocation(loc) {
    this.setState({ latitude: loc.latitude, longitude: loc.longitude });
  }

  render() {
    console.log(this.state)
    return (
      <View style={styles.container}>
        <View style={styles.map}>
          <Map
            setMemoryLocation={this.setMemoryLocation.bind(this)}
          />
        </View>
        <View style={styles.buttonArea}>
          <Button
            small
            backgroundColor="#00BFFF"
            title="record"
            disabled={this.state.disableButton}
          />
        </View>
      </View >
    );
  }
}

const styles = {
  container: {
    flex: 1,
  },
  map: {
    flex: 5,
  },
  buttonArea: {
    flex: 1,
  },
};
