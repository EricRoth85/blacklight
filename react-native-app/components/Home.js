'use strict';
import React, { Component } from 'react';
import { StyleSheet, Text, View, Button } from 'react-native';
import { TabNavigator } from 'react-navigation';
import FrontPage from './FrontPage';

import User from './User';

export default class Home extends Component {


    render() {

      return (
        <View style={styles.container}>
          <Text style={styles.text}>WELCOME TO BLACKLIGHT</Text>
        </View>

      );
    }
  }

  const styles = StyleSheet.create({
    container: {
      flex: 1,
      backgroundColor: 'black',
      alignItems: 'center',
      justifyContent: 'center',
    },
    text: {
      fontSize: 40,
      fontWeight: 'bold',
      color: 'white',
      textAlign: 'center'
    },
    icon: {
      width: 26,
      height: 26,
    }
  });
