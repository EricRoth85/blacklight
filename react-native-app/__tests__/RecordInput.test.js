'use strict';
import React from 'react';
import renderer from 'react-test-renderer';
import { createStore } from 'redux';

import { RecordInput } from '../components';

describe('RecordInput component', () => {
  let store = createStore( () => ({}) );
  it('renders without crashing', () => {
    const RecordInputRenderer = renderer.create(<RecordInput store={store}/>).toJSON();
    expect(RecordInputRenderer).toMatchSnapshot();
  });
})