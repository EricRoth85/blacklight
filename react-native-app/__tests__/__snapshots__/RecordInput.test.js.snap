// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecordInput component renders without crashing 1`] = `
<View
  keyboardVerticalOffset={0}
  onLayout={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "black",
        "flex": 1,
        "justifyContent": "center",
        "padding": 20,
      },
      Object {
        "paddingBottom": 0,
      },
    ]
  }
>
  <View
    style={
      Object {
        "margin": 100,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
          undefined,
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {},
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            Array [
              Object {
                "color": "#86939e",
                "fontSize": 15,
                "fontWeight": "bold",
                "marginBottom": 1,
                "marginLeft": 20,
                "marginRight": 20,
                "marginTop": 15,
              },
              undefined,
              undefined,
            ],
          ]
        }
      >
        Title
      </Text>
    </View>
    <View
      collapsable={undefined}
      style={
        Object {
          "borderBottomColor": "#bdc6cf",
          "borderBottomWidth": 1,
          "marginLeft": 20,
          "marginRight": 20,
          "transform": Array [
            Object {
              "translateX": 0,
            },
          ],
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        color="#FFFFFF"
        onChangeText={[Function]}
        placeholer="Title"
        style={
          Array [
            Object {
              "color": "#86939e",
              "minHeight": 36,
              "width": 750,
            },
            Object {
              "fontSize": 17.5,
            },
            undefined,
          ]
        }
      />
    </View>
    <View
      style={
        Array [
          Object {},
          undefined,
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {},
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            Array [
              Object {
                "color": "#86939e",
                "fontSize": 15,
                "fontWeight": "bold",
                "marginBottom": 1,
                "marginLeft": 20,
                "marginRight": 20,
                "marginTop": 15,
              },
              undefined,
              undefined,
            ],
          ]
        }
      >
        YOUR STORY
      </Text>
    </View>
    <View
      collapsable={undefined}
      style={
        Object {
          "borderBottomColor": "#bdc6cf",
          "borderBottomWidth": 1,
          "marginLeft": 20,
          "marginRight": 20,
          "transform": Array [
            Object {
              "translateX": 0,
            },
          ],
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        color="#FFFFFF"
        multiline={true}
        numberOfLines={7}
        onChangeText={[Function]}
        placeholer="Your Story"
        style={
          Array [
            Object {
              "color": "#86939e",
              "minHeight": 36,
              "width": 750,
            },
            Object {
              "fontSize": 17.5,
            },
            undefined,
          ]
        }
        value=""
      />
    </View>
    <View>
      <View
        style={
          Array [
            Object {
              "marginLeft": 15,
              "marginRight": 15,
            },
            undefined,
            undefined,
            undefined,
          ]
        }
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              Object {
                "backgroundColor": "#000000",
                "padding": 10,
              },
            ]
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            pointerEvents="box-only"
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#9E9E9E",
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "padding": 19,
                },
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                Object {
                  "backgroundColor": "#ffffff",
                },
                undefined,
                Object {
                  "padding": 12,
                },
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
                undefined,
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {},
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  Array [
                    Object {
                      "color": "white",
                      "fontSize": 20,
                    },
                    Object {
                      "color": "#000000",
                    },
                    Object {
                      "fontSize": 17.5,
                    },
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                  ],
                ]
              }
            >
              RECORD
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
